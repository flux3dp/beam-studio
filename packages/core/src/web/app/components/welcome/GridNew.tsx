import React from 'react';

import { PlusOutlined } from '@ant-design/icons';
import classNames from 'classnames';

import useI18n from '@core/helpers/useI18n';

import styles from './GridFile.module.scss';

interface Props {
  startNewProject: () => void;
}

const GridNew = ({ startNewProject }: Props) => {
  const { welcome_page: t } = useI18n();

  return (
    <div className={styles['text-container']}>
      <div className={classNames(styles['text-content'], styles.button)} onClick={startNewProject}>
        <PlusOutlined className={styles.icon} />
        <div className={styles.text}>{t.new_project}</div>
      </div>
    </div>
  );
};

export default GridNew;
