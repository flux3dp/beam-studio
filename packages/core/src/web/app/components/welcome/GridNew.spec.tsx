import React from 'react';

import { fireEvent, render } from '@testing-library/react';

const mockClearScene = jest.fn();

jest.mock('@core/app/actions/beambox/svgeditor-function-wrapper', () => ({
  clearScene: mockClearScene,
}));

import GridNew from './GridNew';

describe('test GridNew', () => {
  it('should render correctly', () => {
    const { container } = render(<GridNew />);

    expect(container).toMatchSnapshot();

    const button = container.querySelector('.button');

    fireEvent.click(button);
    expect(mockClearScene).toHaveBeenCalled();
  });
});
