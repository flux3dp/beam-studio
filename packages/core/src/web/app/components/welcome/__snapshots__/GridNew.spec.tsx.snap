// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GridNew should render correctly 1`] = `
<div>
  <div
    class="text-container"
  >
    <div
      class="text-content button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
      <div
        class="text"
      >
        New Project
      </div>
    </div>
  </div>
</div>
`;
